lib/iris/analysis/_regrid.py

```diff
@@ -452,7 +455,13 @@ def regrid_reference_surface(src_surface_coord, surface_dims,
                 warnings.warn(msg)
         return result
 
-    def _check_units(self, coord):
+    @staticmethod
+    def _check_units(coord):
```

Comment:
My reasoning behind trying to share as much code as possible was that it saved time spent writing unit tests since (hopefully) this method would already be tested.

Do you mean factor it out as in have it as a separate function?